proto/api/v2/report_service.proto

```diff
@@ -7,9 +7,40 @@ import "google/protobuf/timestamp.proto";
 import "api/v2/common.proto";
 import "api/v2/search_query.proto";
 import weak "google/api/annotations.proto";
+import "api/v2/report_configuration_service.proto";
+import "api/v2/user.proto";
 
 package v2;
 
+message GetReportHistoryRequest {
+    string   report_config_id  = 1;
+    RawQuery query             = 2;
```

Comment:
Renaming this field to `report_param_query` would make the meaning clearer. Update the design document as well.